src/mdframe/reader.py

```diff
@@ -27,19 +31,20 @@ def metadata_file_to_df(metadata_file_contents):
 
 
 def data_dir_to_dataframes(data_dir_path: Path,
-                           metadata_file_extension: MetadataFileExtension) -> List[pd.DataFrame]:
-    metadata_file_paths = data_dir_path.glob(f"*.{metadata_file_extension}")
-    metadata_file_contents = map(load_metadata_file, metadata_file_paths)
+                           metadata_file_extension: MetadataFileExtension,
+                           schema: Dict) -> List[pd.DataFrame]:
+    metadata_file_paths = data_dir_path.glob(f"*.{metadata_file_extension}")    
+    metadata_file_contents = [load_metadata_file(path, schema) for path in metadata_file_paths]
     metadata_records = map(metadata_file_to_df, metadata_file_contents)
     return list(metadata_records)
 
-
 @dataclass
 class Config:
     data_dir_path: Path
     metadata_file_extension: Optional[MetadataFileExtension]
     # data_file_extensions: List[DataFileExtension]
-
+    schema: Dict
```

Comment:
You can use a `Dict` here but I would suggest the `schema.py` to be actually dumped into JSON. In this format we can leverage JSONSchema plugins in e.g. VSCode. As for how the dict would be loaded here you could do something like:

```python
from pathlib import Path
from urllib3 import Url


@dataclass
class Config:
    data_dir_path: Path
    metadata_file_extension: Optional[MetadataFileExtension]
    # data_file_extensions: List[DataFileExtension]
    schema_url: Url | Path
    __schema: Dict | None = None

    @property
    def schema(self):
        if self.__schema is None:
            self.__schema = # code to load dict schema from a JSON file
        return self.__schema
```

Reply:
taken care of (distinguished between Url, Path cases)